src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -147,6 +147,10 @@ CHIP_ERROR NrfWiFiDriver::CommitConfiguration()
 
 CHIP_ERROR NrfWiFiDriver::RevertConfiguration()
 {
+    // Abort Connection Recovery if it is in progress during reverting configuration.
+    // This is needed to stop recovery process after failsafe timer expiring.
+    WiFiManager::Instance().AbortConnectionRecovery();
```

Comment:
Shouldn't this simply be called by `WiFiManager::Instance().Disconnect()`? IMHO it seems natural that when I disconnect I really mean it, and the connection shouldn't be resurrected on the next timer fire time :)

Reply:
Indeed it makes total sense to call this in the Disconnect() as well, but here we need to avoid further retires when the fail safe timer expires, so it should not be dependent on the connection state (we call Disconnect() here only when the device is connected to the network), or do I miss something? Maybe we can call Disconnect regardless of the state, that would address all use cases. Or move the check to the Disconnect itself...

Reply:
Ok, so let's leave it for now, but IMHO it would be cleaner if this retry logic was hidden inside WiFiManager so that the caller does not need to know all the dependencies between methods.

Reply:
I agree, this should be the goal. I hope this can be fixed in the next iteration.